src/xpk/core/core.py

```diff
@@ -1200,6 +1198,33 @@ def is_gcsfuse_driver_enabled_on_cluster(args) -> bool:
   return False
 
 
+def is_gcpfilestore_driver_enabled_on_cluster(args) -> bool:
+  """Checks if GCPFilestore CSI driver is enabled on the cluster.
+  Args:
+    args: user provided arguments for running the command.
+  Returns:
+    True if GCPFilestore CSI driver is enabled on the cluster and False otherwise.
+  """
+  command = (
+      f'gcloud container clusters describe {args.cluster}'
+      f' --project={args.project} --region={zone_to_region(args.zone)}'
+      ' --format="value(addonsConfig.gcpFilestoreCsiDriverConfig.enabled)"'
+  )
+  return_code, gcpfilestore_driver_enabled = run_command_for_value(
+      command,
+      'Checks if GCPFilestore CSI driver is enabled in cluster describe.',
+      args,
+  )
+  if return_code != 0:
+    xpk_exit(return_code)
+  if gcpfilestore_driver_enabled.lower() == 'true':
+    xpk_print(
+        'GCPFilestore CSI driver is enabled on the cluster, no update needed.'
+    )
+    return True
+  return False
```

Comment:
Code duplication. Extract to separate function similar to update_gke_cluster_with_addon. Possible signature: is_driver_enabled(driver_config_name: str) -> bool.

Reply:
Actually there are more functions similar to these ones, with varying format of the command output and comments printed out. I would suggest creating additional function in this case:
check_clusster_config(format: str, task: str, positive_msg: str, truth_check) -> bool

Please give me your opinion about this as this is a little bit of a bigger refactor, although still contained within one file.

Reply:
I will refactor to is_driver_enabled please create refactoring task for other places